File: frontend/src/types/BrandingSettings.ts
export interface BrandingSettings {
  logoLight?: string;
  logoDark?: string;
  favicon?: string;

  primaryColor: string;
  secondaryColor?: string;
  accentColor?: string;
  backgroundColor?: string;
  textColor?: string;

  fontFamily?: string;
  fontSize?: string;

  customTheme?: Record<string, any>;

  isActive?: boolean;

  createdAt?: string; // ISO string
  updatedAt?: string; // ISO string
}

export interface CreateBrandingSettingsRequest {
  logoLight?: string;
  logoDark?: string;
  favicon?: string;

  primaryColor: string;
  secondaryColor?: string;
  accentColor?: string;
  backgroundColor?: string;
  textColor?: string;

  fontFamily?: string;
  fontSize?: string;

  customTheme?: Record<string, any>;

  isActive?: boolean;
}

export interface UpdateBrandingSettingsRequest {
  logoLight?: string;
  logoDark?: string;
  favicon?: string;

  primaryColor?: string;
  secondaryColor?: string;
  accentColor?: string;
  backgroundColor?: string;
  textColor?: string;

  fontFamily?: string;
  fontSize?: string;

  customTheme?: Record<string, any>;

  isActive?: boolean;
}

File: backend/src/main/java/nextpos/app/nextpos/model/entity/BrandingSettings.java
package nextpos.app.nextpos.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.Map;

/**
 * Branding and theming configuration for a company.
 * Can store logos, color schemes, typography, and custom UI preferences.
 */
@Entity
@Table(name = "branding_settings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@EntityListeners(AuditingEntityListener.class)
public class BrandingSettings {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;

    // Association to Company
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    // Logos
    @Column(name = "logo_light")
    private String logoLight;

    @Column(name = "logo_dark")
    private String logoDark;

    @Column(name = "favicon")
    private String favicon;

    // Colors
    @Column(name = "primary_color")
    private String primaryColor;

    @Column(name = "secondary_color")
    private String secondaryColor;

    @Column(name = "accent_color")
    private String accentColor;

    @Column(name = "background_color")
    private String backgroundColor;

    @Column(name = "text_color")
    private String textColor;

    // Typography
    @Column(name = "font_family")
    private String fontFamily;

    @Column(name = "font_size")
    private String fontSize;

    // Additional UI customization (stored as JSON)
    @Column(name = "custom_theme", columnDefinition = "jsonb")
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> customTheme;

    // Status flags
    @Column(name = "is_active")
    private Boolean isActive;

    @Column(name = "created_by", nullable = false, updatable = false)
    private Long createdBy;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_by")
    private Long updatedBy;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
